docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
+perform these tasks:
 
-1. KubeStellar
-2. [the status addon](https://github.ibm.com/dettori/status-addon)
-3. KubeFlex
+1. Create *Workload Definition Spaces* (WDSs) to store the definitions of their workloads.
+A Kubernetes workload is an application that runs on Kubernetes. A workload can be made by a 
+single Kubernetes object or several objects that work together.
+2. Create *Inventory and Transport Spaces* (ITSs) to manage the inventory of clusters and 
+the transport of workloads.
+3. Register and label Workload Execution Clusters (WECs) with the Inventory and 
+Transport Space, to keep track of the available clusters and their characteristics.
+4. Define Placement Policies to specify *what" objects and *where* should be 
+deployed on the WECs.
+5. Submit objects in the native Kubernetes format to the WDSs, 
+and let the Placement Policies govern which WECs should receive them.
+6. Check the status of submitted objects from the WDS.
 
-The last two are independent and KubeStellar depends on them.
+In KubeStellar, users can assume a variety of roles and responsibilities. 
+These roles could range from system administrators and application owners 
+to CISOs and DevOps Engineers. However, for the purpose of this document, 
+we will not differentiate between these roles. Instead, we will use the 
+term ‘user’ to collectively refer to any individual who encompasses all these roles.
 
-The status addon defines the `WorkStatus` Kind of objects and some (what?) associated stuff.
+Examples of users interaction with KubeStellar are illustrated in the
+[KubeStellar Usage Examples](./examples.md) section.
 
-## KubeStellar
+The KubeStellar architecture has these main modules:
 
-KubeStellar currently has two major parts.
+- *KubeStellar Controller Manager*: this module is responsible of delivering workload
+objects from the WDS to the ITS according to placement policies, and 
+updating the status of objects in the WDS.
 
-1. OCM for inventory and transport
-2. A central controller-manager that provides the added functionalities
+- *Space Manager*: This module manages the lifecycle of spaces.
 
-We plan to make the transport pluggable in the near future.
+- *OCM Cluster Manager*: This module syncs objects from the ITS to the Workload Execution 
+Clusters (WECs). In the ITS, each mailbox namespace is associated with one WEC. Objects 
+that are put in a mailbox namespace are delivered to the matching WEC.
 
+- *Status Add-On Controller*: This module installs the OCM status add-on agent 
+on all WECs and sets the RBAC permissions for it using the OCM add-on framework.
+
+- *OCM Agent*: This module registers the WEC to the OCM Hub, watches for 
+ManifestWorks and unwraps and syncs the objects into the WEC.
+
+- *OCM Status Add-On Agent*: This module watches *AppliedManifestWorks* 
+to find objects that are synced by the OCM agent, gets their status 
+and updates *workstatuses* objects in the ITS namespace associated with the WEC.
```

Comment:
In English text I would use the Kind rather than the resource name (as, for example, is done for ManifestWork).

Reply:
ok, done for all the occurrences in the text.

Reply:
Thanks. There is still one on line 74.

Reply:
fixed